### PR TITLE
Update async examples

### LINKED ISSUE #9874: avoid compiler warnings ("wolf, wolf")
earlier stated that marking a method with async tag but not having an await is not wrong but compiler spits out a warning.
OK but I don't think the code should encourage such noise so my beef is with this line in the "Complete example" listing
async Task Task_MethodAsync()

so imho you should drop that worthless async token here

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 17d27a0d-c072-8c9f-7e31-3de60bee9ae8
* Version Independent ID: e0c09794-e712-19ff-cb9f-12648d60d4f6
* Content: [Asynchronous Programming with async and await (C#)](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/async/)
* Content Source: [docs/csharp/programming-guide/concepts/async/index.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/concepts/async/index.md)
* Product: **dotnet-csharp**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### LINKED ISSUE #9875: method call incomplete without input parameter
in "Async methods are easier to write" section: collapsing the earlier lines into
string urlContents = await client.GetStringAsync();
is incomplete as you still need the string/Uri parameter, so should read
string urlContents = await client.GetStringAsync("https://msdn.microsoft.com");

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 17d27a0d-c072-8c9f-7e31-3de60bee9ae8
* Version Independent ID: e0c09794-e712-19ff-cb9f-12648d60d4f6
* Content: [Asynchronous Programming with async and await (C#)](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/async/)
* Content Source: [docs/csharp/programming-guide/concepts/async/index.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/concepts/async/index.md)
* Product: **dotnet-csharp**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by nxtn**
Fixes #9875 
Fixes #9874 

Contributes to #9781 

Remove redundant references

### CODE REVIEW

File: docs/csharp/programming-guide/concepts/async/index.md [hunk 1]
```diff
@@ -144,9 +145,9 @@ The `async` and `await` keywords don't cause additional threads to be created. A
 The async-based approach to asynchronous programming is preferable to existing approaches in almost every case. In particular, this approach is better than the <xref:System.ComponentModel.BackgroundWorker> class for I/O-bound operations because the code is simpler and you don't have to guard against race conditions. In combination with the <xref:System.Threading.Tasks.Task.Run%2A?displayProperty=nameWithType> method, async programming is better than <xref:System.ComponentModel.BackgroundWorker> for CPU-bound operations because async programming separates the coordination details of running your code from the work that `Task.Run` transfers to the threadpool.  
   
 ##  <a name="BKMK_AsyncandAwait"></a> async and await  
- If you specify that a method is an async method by using the [async](../../../../csharp/language-reference/keywords/async.md) modifier, you enable the following two capabilities.
```
Comment (BillWagner) [inline docs/csharp/programming-guide/concepts/async/index.md]: I'd like to keep the links to the keywords here.

File: docs/csharp/programming-guide/concepts/async/index.md [hunk 2]
```diff
@@ -144,9 +145,9 @@ The `async` and `await` keywords don't cause additional threads to be created. A
...
   
 ##  <a name="BKMK_AsyncandAwait"></a> async and await  
- If you specify that a method is an async method by using the [async](../../../../csharp/language-reference/keywords/async.md) modifier, you enable the following two capabilities.  
+ If you specify that a method is an async method by using the `async` modifier, you enable the following two capabilities.  
   
--   The marked async method can use [await](../../../../csharp/language-reference/keywords/await.md) to designate suspension points. The `await` operator tells the compiler that the async method can't continue past that point until the awaited asynchronous process is complete. In the meantime, control returns to the caller of the async method.  
+-   The marked async method can use `await` to designate suspension points. The `await` operator tells the compiler that the async method can't continue past that point until the awaited asynchronous process is complete. In the meantime, control returns to the caller of the async method.
```
Comment (BillWagner) [inline docs/csharp/programming-guide/concepts/async/index.md]: Same comment: Let's preserve the links.

**Review by BillWagner — CHANGES_REQUESTED** [review]
Overall, this is a great set of changes @NextTurn 

I'd like to keep the links to the keyword references for `async` and `await` (that way, these samples don't have to define those in detail).

If you make that update, I'll :shipit:

**Comment by NextTurn** [comment]
I assume that the following references are redundant

https://github.com/dotnet/docs/blob/f46493f2c26510ebfdd4567eaf5e72613d8d56dc/docs/csharp/programming-guide/concepts/async/index.md#L158-L163

**Comment by NextTurn** [comment]
Hi @BillWagner, don't forget this.

**Comment by NextTurn** [comment]
I assume that the following references are redundant

https://github.com/dotnet/docs/blob/f46493f2c26510ebfdd4567eaf5e72613d8d56dc/docs/csharp/programming-guide/concepts/async/index.md#L158-L162

**Comment by NextTurn** [comment]
@BillWagner

**Comment by NextTurn** [comment]
Welcome back from London @BillWagner

**Review by BillWagner — APPROVED** [review]

**Pull request merged**